src/journal/compress.c

```diff
@@ -534,13 +683,92 @@ int compress_stream_lz4(int fdf, int fdt, uint64_t max_bytes) {
                   total_in, total_out,
                   (double) total_out / total_in * 100);
  cleanup:
-        munmap(src, st.st_size);
+        (void)munmap(src, st.st_size);
         return r;
 #else
         return -EPROTONOSUPPORT;
 #endif
 }
 
+int compress_stream_zstd(int fdf, int fdt, uint64_t max_bytes) {
+#if HAVE_ZSTD
+        _cleanup_(ZSTD_freeCCtxp) ZSTD_CCtx *s = ZSTD_createCStream();
```

Comment:
please don't mix declaring variables with actual code. Please let's keep that separate. See CODING_STYLE about this too. Please declare variables up-top, and then let the code (i.e. the actual invocation of a function like ZSTD_createCStream()) follow after that. Initializing variables while declaring them is fine and welcome, but please only to fixed values, not to actual code.